joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -108,17 +123,37 @@ class JointStatePublisher():
                     zeroval = 0
 
                 joint = {'min': minval, 'max': maxval, 'zero': zeroval}
-                if pub_def_positions:
+                if self.pub_def_positions:
                     joint['position'] = zeroval
-                if pub_def_vels:
+                if self.pub_def_vels:
                     joint['velocity'] = 0.0
-                if pub_def_efforts:
+                if self.pub_def_efforts:
                     joint['effort'] = 0.0
 
                 if jtype == 'continuous':
                     joint['continuous'] = True
                 self.free_joints[name] = joint
 
+    def __init__(self):
+        description = get_param('robot_description')
+
+        self.free_joints = {}
+        self.joint_list = [] # for maintaining the original order of the joints
+        self.dependent_joints = get_param("dependent_joints", {})
+        self.use_mimic = get_param('use_mimic_tags', True)
+        self.use_small = get_param('use_smallest_joint_limits', True)
+
+        self.zeros = get_param("zeros")
+
+        self.pub_def_positions = get_param("publish_default_positions", True)
+        self.pub_def_vels = get_param("publish_default_velocities", False)
+        self.pub_def_efforts = get_param("publish_default_efforts", False)
+
+        if xml.dom.minidom.parseString(description).getElementsByTagName('COLLADA'):
```

Comment:
It looks like you need to access the xml DOM from `__init__()`, `init_urdf()`, and `init_collada()`.  Instead of reparsing the XML twice, I think it would make more sense to parse it here, then pass it into `init_collada()` and `init_urdf()`, like:

```
robot = xml.dom.mindom.parseString(description)
if robot.getElementsByTagName('COLLADA'):
    self.init_collada(robot)
else:
    self.init_urdf(robot)
```

Reply:
ok, done